src/HelperPluginManager.php

```diff
@@ -171,30 +169,29 @@ public function __construct(
      * Validates against callables and HelperInterface implementations.
      *
      * @throws InvalidServiceException
-     * @psalm-assert HelperInterface|callable $instance
+     * @psalm-assert callable $instance
      */
     public function validate(mixed $instance): void
     {
-        if (! is_callable($instance) && ! $instance instanceof HelperInterface) {
+        if (! is_callable($instance)) {
             throw new InvalidServiceException(
                 sprintf(
-                    '%s can only create instances of %s and/or callables; %s is invalid',
+                    '%s can only create callables; %s is invalid',
                     self::class,
-                    HelperInterface::class,
                     get_debug_type($instance),
                 ),
             );
         }
     }
 
     /**
-     * @template InstanceParam of HelperInterface
+     * @template InstanceParam
      * @param class-string<InstanceParam>|string $id Service name of plugin to retrieve.
-     * @return ($id is class-string<InstanceParam> ? InstanceParam : InstanceType)
+     * @return ($id is class-string<InstanceParam> ? InstanceParam : callable)
```

Comment:
Will this work?

```suggestion
     * @return ($id is class-string<InstanceParam> ? InstanceParam&callable : callable)
```

Reply:
It does not matter. It needs to be on the interface to be useful

Reply:
Forgot the interface… done.